### PR TITLE
feat: add contributing style guide

### PR DESCRIPTION
**Pull request by SuperFluffy**
## Summary
Add a first style guide for how to write services at Astria.

## Background
With Astria ramping up and getting more contributors we need to formalize how we write services. With this repository being Rust, this document focuses on Rust, observability, and error handling. 

## Changes
* Add a CONTRIBUTING.md

### CODE REVIEW

File: CONTRIBUTING.md
Comment (SuperFluffy) [inline CONTRIBUTING.md:1]: This might be better named `CODING-GUIDELINES.md` I am not sure what the best way to call this or where to put it. @joroshiba

Comment (joroshiba) [reply CONTRIBUTING.md:1]: Spec folder is probably appropriate? We can have a seperate contributing guidelines around things like PRs etc which links to it.

**Comment by SuperFluffy** [comment]
Add section on:

1. config
2. creating services, actors, singleton pattern
3. builder

**Comment by joroshiba** [comment]
This PR is stale because it has been open 45 days with no activity. Remove stale label or this PR will be
closed in 7 days.

**Comment by joroshiba** [comment]
This PR was closed because it has been stale.

**Pull request closed without merging**